static/js/containers/SettingsPage.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+import Loader from 'react-loader';
+import Grid, { Cell } from 'react-mdl/lib/Grid';
+
+import Jumbotron from '../components/Jumbotron';
+import {
+  startProfileEdit,
+  FETCH_PROCESSING,
+} from '../actions/index';
+import ProfileFormContainer from './ProfileFormContainer';
+import PrivacyForm from '../components/PrivacyForm';
+import ProfileProgressControls from '../components/ProfileProgressControls';
+import {
+    combineValidators,
+    privacyValidation,
+} from '../util/validation';
+import { getPreferredName } from '../util/util';
+import type { Profile } from '../flow/profileTypes';
+import type { UIState } from '../reducers/ui';
+
+class SettingsPage extends ProfileFormContainer {
+  props: {
+    profiles: {[key: string]: Profile},
+    dispatch: Function,
+    ui:       UIState,
+  };
+
+  componentDidMount() {
+    this.fetchProfile(SETTINGS.username);
+  }
+
+  componentDidUpdate() {
+    this.fetchProfile(SETTINGS.username);
+  }
```

Comment:
Sorry, I just noticed this. I deleted these two methods (`componentDidUpdate` and `componentDidMount`) and it seems like everything is taken care of by `App`, which also calls these methods. 

Can you:

- delete those two methods
- add a default value to `profile` (e.g. `profile = {};`) in the render method, so that if the profile GET request hasn't come back yet we have something to pass to `ProfileProgressControls`.

sorry! I should have realized that earlier >_M